src/builders/sapi.rs

```diff
@@ -0,0 +1,88 @@
+use crate::ffi::sapi_header_struct;
+use crate::{embed::SapiModule, error::Result};
+
+use std::ffi::c_void;
+use std::{ffi::CString, ptr};
+
+pub struct SapiBuilder {
+    name: String,
+    pretty_name: String,
+    module: SapiModule,
+}
+
+impl SapiBuilder {
+    pub fn new<T: Into<String>, U: Into<String>>(name: T, pretty_name: U) -> Self {
+        Self {
+            name: name.into(),
+            pretty_name: pretty_name.into(),
+            module: SapiModule {
+                name: ptr::null_mut(),
+                pretty_name: ptr::null_mut(),
+                startup: None,
+                shutdown: None,
+                activate: None,
+                deactivate: None,
+                ub_write: None,
+                flush: None,
+                get_stat: None,
+                getenv: None,
+                sapi_error: None,
+                header_handler: None,
+                send_headers: None,
+                send_header: None,
+                read_post: None,
+                read_cookies: None,
+                register_server_variables: None,
+                log_message: None,
+                get_request_time: None,
+                terminate_process: None,
+                php_ini_path_override: ptr::null_mut(),
+                default_post_reader: None,
+                treat_data: None,
+                executable_location: ptr::null_mut(),
+                php_ini_ignore: 0,
+                php_ini_ignore_cwd: 0,
+                get_fd: None,
+                force_http_10: None,
+                get_target_uid: None,
+                get_target_gid: None,
+                input_filter: None,
+                ini_defaults: None,
+                phpinfo_as_text: 0,
+                ini_entries: ptr::null_mut(),
+                additional_functions: ptr::null(),
+                input_filter_init: None,
+            },
+        }
+    }
+
+    /// Sets the send header function for this SAPI
+    ///
+    /// # Arguments
+    ///
+    /// * `func` - The function to be called on shutdown.
+    pub fn send_header_function(mut self, func: SapiSendHeaderFunc) -> Self {
+        self.module.send_header = Some(func);
+        self
+    }
```

Comment:
a lot of thing can be defined but ATM this is the only required function for PHP to not segfault (that's why we set a dummy handler if nothing is set latter)

Reply:
That's enough for me, let's make it evolve when users will have more usage